File: Aegis/Assets/Scripts/Camera/CameraScroll.js
#pragma strict

var startPoint : GameObject;
var endPoint : GameObject;
var scrollSpeed : float;

private var start : float = 0;
private var end : float = 0;
private var timeStarted : float = 0;
function Start () {
	timeStarted = Time.time;
	start = startPoint.transform.position.x;
	end = endPoint.transform.position.x;
}

function Update () {
	var distanceCovered = (Time.time - timeStarted) * scrollSpeed;
	var fracJourney = distanceCovered / end;
	transform.position.x = Mathf.Lerp(start, end, fracJourney);
}

File: Aegis/Assets/Scripts/Camera/CameraScroll.cs
using UnityEngine;
using System.Collections;

public class CameraScroll : MonoBehaviour 
{
    public GameObject startPoint;
    public GameObject endPoint;
    public float scrollSpeed;

    private float start = 0;
    private float end = 0;
    private float timeStarted = 0;
    
    void Start () 
    {
	    timeStarted = Time.time;
	    start = startPoint.transform.position.x;
	    end = endPoint.transform.position.x;
    }

    void Update()
    {
        var distanceCovered = (Time.time - timeStarted) * scrollSpeed;
        var fracJourney = distanceCovered / end;
        transform.position = new Vector3(Mathf.Lerp(start, end, fracJourney), transform.position.y, transform.position.z);
    }
}
